testcontainers/src/core/client.rs

```diff
@@ -421,6 +421,13 @@ impl Client {
     }
 
     pub(crate) async fn docker_hostname(&self) -> Result<url::Host, ClientError> {
+        // allow TESTCONTAINERS_HOST_OVERRIDE to override the docker host
+        if let Some(host_override) =
+            <env::Os as env::GetEnvValue>::get_env_value("TESTCONTAINERS_HOST_OVERRIDE")
+        {
+            return url::Host::parse(&host_override)
+                .map_err(|_| ConfigurationError::InvalidDockerHost(host_override).into());
+        }
```

Comment:
There is more appropriate place for this - https://github.com/testcontainers/testcontainers-rs/blob/d652b383ae75c0b1e56362751e4fb8a7b80d8bb2/testcontainers/src/core/env/config.rs#L127-L137

Moreover config already has `tc_host` which is prioritized, but we read it only from properties file. 
What we can do is to extend [this part](https://github.com/testcontainers/testcontainers-rs/blob/d652b383ae75c0b1e56362751e4fb8a7b80d8bb2/testcontainers/src/core/env/config.rs#L103-L119) and read tc_host from env variables too